pandas/core/groupby/groupby.py

```diff
@@ -489,6 +489,21 @@ def group_selection_context(groupby: "BaseGroupBy") -> Iterator["BaseGroupBy"]:
 class BaseGroupBy(PandasObject, SelectionMixin, Generic[FrameOrSeries]):
     _group_selection: Optional[IndexLabel] = None
     _apply_allowlist: FrozenSet[str] = frozenset()
+    _hidden_attrs = PandasObject._hidden_attrs | {
+        "as_index",
```

Comment:
yeah can you differentiate between deprecations that we are hiding and attributes that we are hing for another reason (e.g. these are private)

Reply:
There may be deprecated attributes, that should not be hidden and vica versa. Also the deprecation log is in #30228, if we want to list deprecated attributes.

IMO it's ok to keep the concepts of hidden and deprecated attributes seperate.